tools/project_manager/show_good_word_suggestions.php

```diff
@@ -15,38 +15,58 @@
 set_time_limit(0); // no time limit
 
 $projectid  = validate_projectID('projectid', @$_REQUEST['projectid']);
-$fileObject = get_project_word_file($projectid,"good");
-$timeCutoff = get_integer_param($_REQUEST, 'timeCutoff', $fileObject->mod_time, 0, null);
 $freqCutoff = get_integer_param($_REQUEST, 'freqCutoff', 5, 0, null);
 
 enforce_edit_authorization($projectid);
 
-if($timeCutoff==0)
-    $time_cutoff_text = _("<b>All proofreader suggestions</b> are included in the results.");
-else
-    $time_cutoff_text = sprintf(_("Only proofreader suggestions made <b>after %s</b> are included in the results."),strftime($datetime_format,$timeCutoff));
-
-
 // $format determines what is presented from this page:
 //   'html' - page is rendered with frequencies included
 //   'file' - all words and frequencies are presented as a
 //            downloaded file
 // 'update' - update the list
 $format = get_enumerated_param($_REQUEST, 'format', 'html', array('html', 'file', 'update'));
+$destination_list = isset($_POST[BAD_WORD_LIST]) ? BAD_WORD_LIST : GOOD_WORD_LIST;
 
 if($format=="update") {
-    $postedWords = parse_posted_words($_POST);
+    if (isset($_POST[REJECT_SUGGESTIONS]))
+    {
+        touch_project_good_words($projectid);
+    }
+    else
+    {
+        $postedWords = parse_posted_words($_POST);
 
-    $words = load_project_good_words($projectid);
-    $words = array_merge($words,$postedWords);
-    save_project_good_words($projectid,$words);
+        $words = $destination_list == GOOD_WORD_LIST ? load_project_good_words($projectid) : load_project_bad_words($projectid);
+        $words = array_merge($words, $postedWords);
+        if ($destination_list == GOOD_WORD_LIST)
+        {
+            save_project_good_words($projectid, $words);
+        }
+        else
+        {
+            save_project_bad_words($projectid, $words);
+        }
+    }
 
     $format="html";
 }
 
+$fileObject = get_project_word_file($projectid,"good");
+$timeCutoff = get_integer_param($_REQUEST, 'timeCutoff', $fileObject->mod_time, 0, null);
+
+if($timeCutoff==0)
+    $time_cutoff_text = _("<b>All proofreader suggestions</b> are included in the results.");
+else
+    $time_cutoff_text = sprintf(_("Only proofreader suggestions made <b>after %s</b> are included in the results."),strftime($datetime_format,$timeCutoff));
+
 list($all_suggestions_w_freq,$all_suggestions_w_occurrences,$round_suggestions_w_freq,$round_suggestions_w_occurrences,$rounds,$round_page_count,$messages) =
     _get_word_list($projectid,$timeCutoff);
 
+if ($destination_list == BAD_WORD_LIST && $timeCutoff == $fileObject->mod_time && $all_suggestions_w_occurrences == 0)
```

Comment:
You're missing a `count()` on the array:
```
if ($destination_list == BAD_WORD_LIST && $timeCutoff == $fileObject->mod_time && count($all_suggestions_w_occurrences) == 0)
```

Reply:
The following diff will do effectively the same thing for the `show_all_good_word_suggestions.php` file:
```diff
diff --git a/tools/project_manager/show_all_good_word_suggestions.php b/tools/project_manager/show_all_good_word_suggestions.php
index 34fc53f..db5d9e4 100644
--- a/tools/project_manager/show_all_good_word_suggestions.php
+++ b/tools/project_manager/show_all_good_word_suggestions.php
@@ -186,7 +186,17 @@ echo "<br>";

         // if no words are returned (probably because something was
         // suggested but is no longer in the text) skip this project
-        if(count($suggestions_w_freq)==0) continue;
+        if(count($suggestions_w_freq)==0)
+        {
+            // If the time cutoff is the same as the good words file and
+            // we still have no suggestions, touch the good words file to
+            // reset the time so no alert appears for this project.
+            if($timeCutoffActual == $goodFileObject->mod_time)
+            {
+                touch_project_good_words($projectid);
+            }
+            continue;
+        }

         $projectsNeedingAttention++;
```